src/sentry/runner/commands/devservices.py

```diff
@@ -46,6 +46,26 @@ def get_or_create(client, thing, name):
         return getattr(client, thing + "s").create(name)
 
 
+def retryable_pull(client, image, max_attempts=5):
+    from docker.errors import ImageNotFound
+
+    current_attempt = 0
+
+    # `client.images.pull` intermittently fails in CI, and the docker API/docker-py does not give us the relevant error message (i.e. it's not the same error as running `docker pull` from shell)
+    # As a workaround, let's retry when we hit the ImageNotFound exception.
+    #
+    # See https://github.com/docker/docker-py/issues/2101 for more information
+    while True:
+        try:
+            client.images.pull(image)
+        except ImageNotFound as e:
+            if current_attempt + 1 >= max_attempts:
+                raise e
+            current_attempt = current_attempt + 1
+            continue
+        break
```

Comment:
Would it be good if we raise something if we run out of attempts? Or log an error?

Reply:
That's what https://github.com/getsentry/sentry/pull/29691/files#diff-1125ff5a13bc41637ad61c8bd8edbc84dc7a8436b8a6ad3a8b95f16d65156913R63 does

Reply:
Than you, sir! I just had a moment :)